sopel/irc/backends.py

```diff
@@ -213,6 +213,8 @@ def collect_incoming_data(self, data):
         The incoming line is discarded (and thus ignored) if guessing the text
         encoding and decoding it fails.
         """
+        data = data + self.get_terminator()
```

Comment:
The problem is now that when you do `self.buffer += data`, `self.buffer` now contains `\r\n` while it didn't before.

Reply:
That's ~intentional. I don't *think* the buffer goes anywhere but the log and the PreTrigger, and `PreTrigger.__init__()` immediately does `line = line.strip('\r\n')`. Everything gracefully handled the previous behavior, which unintentionally left a `\r` at the end of each message.

Reply:
Thinking about this. Indeed. It works because down the line, it's already handled. Thanks for your clarification!